File: Client/php/ApiClient.php
<?php

namespace ApiSdk;

use ApiSdk\EscapedList\ListRequestBuilder;
use ApiSdk\GetData\GetDataRequestBuilder;
use ApiSdk\Search\SearchRequestBuilder;
use Microsoft\Kiota\Abstractions\ApiClientBuilder;
use Microsoft\Kiota\Abstractions\BaseRequestBuilder;
use Microsoft\Kiota\Abstractions\RequestAdapter;
use Microsoft\Kiota\Serialization\Json\JsonParseNodeFactory;
use Microsoft\Kiota\Serialization\Json\JsonSerializationWriterFactory;
use Microsoft\Kiota\Serialization\Text\TextParseNodeFactory;
use Microsoft\Kiota\Serialization\Text\TextSerializationWriterFactory;

/**
 * The main entry point of the SDK, exposes the configuration and the fluent API.
*/
class ApiClient extends BaseRequestBuilder 
{
    /**
     * The list property
    */
    public function escapedList(): ListRequestBuilder {
        return new ListRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * The getData property
    */
    public function getData(): GetDataRequestBuilder {
        return new GetDataRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * The search property
    */
    public function search(): SearchRequestBuilder {
        return new SearchRequestBuilder($this->pathParameters, $this->requestAdapter);
    }
    
    /**
     * Instantiates a new ApiClient and sets the default values.
     * @param RequestAdapter $requestAdapter The request adapter to use to execute the requests.
    */
    public function __construct(RequestAdapter $requestAdapter) {
        parent::__construct($requestAdapter, [], '{+baseurl}');
        ApiClientBuilder::registerDefaultSerializer(JsonSerializationWriterFactory::class);
        ApiClientBuilder::registerDefaultSerializer(TextSerializationWriterFactory::class);
        ApiClientBuilder::registerDefaultDeserializer(JsonParseNodeFactory::class);
        ApiClientBuilder::registerDefaultDeserializer(TextParseNodeFactory::class);
        if (empty($this->requestAdapter->getBaseUrl())) {
            $this->requestAdapter->setBaseUrl('https://api.genome.ucsc.edu/list/publicHubs');
        }
        $this->pathParameters['baseurl'] = $this->requestAdapter->getBaseUrl();
    }

}

File: Client/ruby/api_client.rb
require 'microsoft_kiota_abstractions'
require 'microsoft_kiota_serialization_json'
require_relative './api_sdk'
require_relative './get_data/get_data_request_builder'
require_relative './list/list_request_builder'
require_relative './search/search_request_builder'

module ApiSdk
    ## 
    # The main entry point of the SDK, exposes the configuration and the fluent API.
    class ApiClient < MicrosoftKiotaAbstractions::BaseRequestBuilder
        
        ## 
        # The getData property
        def get_data()
            return ApiSdk::GetData::GetDataRequestBuilder.new(@path_parameters, @request_adapter)
        end
        ## 
        # The list property
        def list()
            return ApiSdk::List::ListRequestBuilder.new(@path_parameters, @request_adapter)
        end
        ## 
        # The search property
        def search()
            return ApiSdk::Search::SearchRequestBuilder.new(@path_parameters, @request_adapter)
        end
        ## 
        ## Instantiates a new ApiClient and sets the default values.
        ## @param request_adapter The request adapter to use to execute the requests.
        ## @return a void
        ## 
        def initialize(request_adapter)
            super(Hash.new, request_adapter, "{+baseurl}")
            MicrosoftKiotaAbstractions::ApiClientBuilder.register_default_serializer(MicrosoftKiotaSerializationJson::JsonSerializationWriterFactory)
            MicrosoftKiotaAbstractions::ApiClientBuilder.register_default_deserializer(MicrosoftKiotaSerializationJson::JsonParseNodeFactory)
            if @request_adapter.get_base_url.nil? || @request_adapter.get_base_url.empty?
                @request_adapter.set_base_url('https://api.genome.ucsc.edu/list/publicHubs')
            end
            @path_parameters['baseurl'] = @request_adapter.get_base_url
        end
    end
end
